exporter/exporterhelper/internal/bounded_memory_queue.go

```diff
@@ -37,17 +37,22 @@ func (q *boundedMemoryQueue[T]) Offer(ctx context.Context, req T) error {
 	}
 
 	select {
-	case q.items <- newQueueRequest(ctx, req):
+	case q.items <- queueRequest[T]{ctx: ctx, req: req}:
 		return nil
 	default:
 		return ErrQueueIsFull
 	}
 }
 
-// Poll returns a request from the queue once it's available. It returns false if the queue is stopped.
-func (q *boundedMemoryQueue[T]) Poll() (QueueRequest[T], bool) {
+// Consume applies the provided function on the head of queue. The call blocks until there is an item available.
```

Comment:
// The call blocks until there is an item available.

// The call blocks until there is an item available or the queue is stopped.